### PR TITLE
IC: Compute symbols impacted by classpath changes

### PR DESCRIPTION
**Pull request by hungvietnguyen**

### CODE REVIEW

File: libraries/tools/kotlin-gradle-plugin-integration-tests/src/test/kotlin/org/jetbrains/kotlin/gradle/IncrementalCompilationMultiProjectIT.kt [hunk 1]
```diff
@@ -776,6 +775,60 @@ abstract class BaseIncrementalCompilationMultiProjectIT : IncrementalCompilation
...
+                    // KT-56197 was fixed for the new IC only, the old IC still doesn't recompile "foo/ClassUsingSubclassInApp.kt"
+                    assertIncrementalCompilation(
+                        expectedCompiledKotlinFiles = getExpectedKotlinSourcesForDefaultProject(
+                            libSources = listOf("bar/InterfaceInLib.kt"),
+                            appSources = listOf("foo/SubclassInApp.kt")
+                        )
+                    )
+                } else {
+                    assertIncrementalCompilation(
+                        expectedCompiledKotlinFiles = getExpectedKotlinSourcesForDefaultProject(
```
Comment (gavra0) [inline libraries/tools/kotlin-gradle-plugin-integration-tests/src/test/kotlin/org/jetbrains/kotlin/gradle/IncrementalCompilationMultiProjectIT.kt:822]: Is ClassUsingSubclassInApp missing?

Comment (hungvietnguyen) [reply libraries/tools/kotlin-gradle-plugin-integration-tests/src/test/kotlin/org/jetbrains/kotlin/gradle/IncrementalCompilationMultiProjectIT.kt:822]: It's included in `appSources` at line 824 below (the list has 2 files).

File: compiler/incremental-compilation-impl/src/org/jetbrains/kotlin/incremental/IncrementalJvmCompilerRunner.kt [hunk 2]
```diff
@@ -299,20 +294,14 @@ open class IncrementalJvmCompilerRunner(
             is NotAvailableForJSCompiler -> error("Unexpected type for this code path: ${classpathChanges.javaClass.name}.")
         }
 
-        @Suppress("UNUSED_VARIABLE") // for sealed when
-        val unused = when (classpathChanges) {
```
[the remark this answers is not in the document]
Comment (hungvietnguyen) [reply compiler/incremental-compilation-impl/src/org/jetbrains/kotlin/incremental/IncrementalJvmCompilerRunner.kt:303]: I think so too, but `unused` looks slightly clumsy. I'm now using a utility function: `when { ... }.forceExhaustiveWhen()`. If you prefer the original version, pls let me know.

**Review by gavra0 — COMMENTED** [review]
LGTM

**Comment by hungvietnguyen** [comment]
Added a commit to fix failed tests.

File: libraries/tools/kotlin-gradle-plugin-integration-tests/src/test/kotlin/org/jetbrains/kotlin/gradle/IncrementalCompilationMultiProjectIT.kt (hunk 1 above)
Comment (Mellobrainbox) [reply libraries/tools/kotlin-gradle-plugin-integration-tests/src/test/kotlin/org/jetbrains/kotlin/gradle/IncrementalCompilationMultiProjectIT.kt]: Thanks, your welcome sir

**Pull request merged**